samcli/lib/bootstrap/companion_stack/companion_stack_builder.py

```diff
@@ -0,0 +1,130 @@
+"""
+    Companion stack template builder
+"""
+from typing import Dict
+
+# pylint: disable=W0402
+from string import Template
+
+from samcli.lib.bootstrap.companion_stack.data_types import CompanionStack, ECRRepo
+from samcli import __version__ as VERSION
+
+_STACK_TEMPLATE = Template(
+    """
+AWSTemplateFormatVersion : '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: AWS SAM CLI Managed ECR Repo Stack
+Metadata:
+  SamCliInfo: $sam_cli_version
+  CompanionStackname:  $companion_stack_name
+
+Resources:
+$resources
```

Comment:
Using string interpolation is fragile to changes, as it's both indentation-sensitive and cannot be easily syntax-checked. Would using Python objects and serializing to JSON help?

Reply:
Good point! Updated to use dict + json.dumps.